src/main/java/org/reaktivity/nukleus/http_cache/internal/proxy/cache/Cache.java

```diff
@@ -136,34 +136,61 @@ public void put(
         CacheEntry oldCacheEntry = cachedEntries.get(requestUrlHash);
         if (oldCacheEntry == null)
         {
-            CacheEntry cacheEntry = new CacheEntry(
+            final CacheEntry cachedEntry;
+
+            if (request.protocolStackHeader())
+            {
+                cachedEntry = new EmulatedCacheEntry(
+                    this,
+                    request,
+                    true,
+                    supplyTrace);
+            }
+            else
+            {
+                cachedEntry = new NativeCacheEntry(
```

Comment:
Please eliminate the word `native` from any of this code, since later we'll remove all the `emulated` stuff and still having `native` around won't make sense then.